app/src/main/java/com/github/libretube/helpers/PreferenceHelper.kt

```diff
@@ -30,6 +37,32 @@ object PreferenceHelper {
         authSettings = getAuthenticationPreferences(context)
     }
 
+    /**
+     * Migrate preference to a new version
+     */
+    fun migrate() {
+        val prefVersion = getInt(PreferenceKeys.PREFERENCE_VERSION, -1)
+        // check if there are any prefs to migrate
+        if (prefVersion == BuildConfig.VERSION_CODE)
+            return
+
+        if (prefVersion < 63) {
+            Log.i(TAG, "Migration to prefs v63")
+            LibreTubeApp.instance.resources
+                .getStringArray(R.array.sponsorBlockSegments)
+                .forEach { category ->
+                    val key = "${category}_category"
+                    val stored = getString(key, "visible")
+                    if (stored == "visible") {
+                        putString(key, SbSkipOptions.MANUAL.name.lowercase())
+                    }
+                }
+        }
+
+        // mark as successfully migrated
+        putInt(PreferenceKeys.PREFERENCE_VERSION, BuildConfig.VERSION_CODE)
```

Comment:
Not sure if it's better to use the version code here, or a separate constant, to avoid having the migrations run on every startup for builds where the version is not increased on every update, e.g. Debug/Nightly builds?

Reply:
I think it's better to manually increment the number, because not every version will have a preference migration, so it'd run the whole function body unnecessarily with each update even if there's no migration necessary.

Depending on how much we want to over-engineer it: In theory we could build a simple `PreferenceMigration` class similar to the room migrations and maintain a list of migrations numbered from 1 upwards, see https://github.com/libre-tube/LibreTube/blob/master/app/src/main/java/com/github/libretube/db/AppDatabase.kt. We could then have a `PreferenceMigration(1, 2) { override fun onMigrate() {} }`, that runs the migration.

Reply:
Implemented in https://github.com/libre-tube/LibreTube/pull/7434/commits/c746647041515ed8e52077753978fb099a53a543.

Reply:
Awesome, thank you!